pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -1304,6 +1340,113 @@ func (a *ApplicationBackupController) backupResources(
 		return err
 	}
 
+	if nfs {
+		// Check whether ResourceExport is preset or not
+		crName := getResourceExportCRName(utils.PrefixBackup, string(backup.UID), backup.Namespace)
+		resourceExport, err := kdmpShedOps.Instance().GetResourceExport(crName, a.backupAdminNamespace)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				// create resource export CR
+				resourceExport := &kdmpapi.ResourceExport{}
+				// Adding required label for debugging
+				labels := make(map[string]string)
+				labels[utils.ApplicationBackupCRNameKey] = utils.GetValidLabel(backup.Name)
+				labels[utils.ApplicationBackupCRUIDKey] = utils.GetValidLabel(utils.GetShortUID(string(backup.UID)))
+				// If backup from px-backup, update the backup object details in the label
+				if val, ok := backup.Annotations[utils.PxbackupAnnotationCreateByKey]; ok {
+					if val == utils.PxbackupAnnotationCreateByValue {
+						labels[utils.BackupObjectNameKey] = utils.GetValidLabel(backup.Annotations[utils.PxbackupObjectNameKey])
+						labels[utils.BackupObjectUIDKey] = utils.GetValidLabel(backup.Annotations[utils.PxbackupObjectUIDKey])
+					}
+				}
+				resourceExport.Labels = labels
+				resourceExport.Annotations = make(map[string]string)
+				resourceExport.Annotations[utils.SkipResourceAnnotation] = "true"
+				resourceExport.Name = getResourceExportCRName(utils.PrefixBackup, string(backup.UID), backup.Namespace)
+				resourceExport.Namespace = a.backupAdminNamespace
+				resourceExport.Spec.Type = kdmpapi.ResourceExportBackup
+				source := &kdmpapi.ResourceExportObjectReference{
+					APIVersion: backup.APIVersion,
+					Kind:       backup.Kind,
+					Namespace:  backup.Namespace,
+					Name:       backup.Name,
+				}
+				backupLocation, err := storkops.Instance().GetBackupLocation(backup.Spec.BackupLocation, backup.Namespace)
+				if err != nil {
+					return fmt.Errorf("error getting backup location path: %v", err)
+				}
+				destination := &kdmpapi.ResourceExportObjectReference{
+					// TODO: .GetBackupLocation is not returning APIVersion and kind.
+					// Hardcoding for now.
+					// APIVersion: backupLocation.APIVersion,
+					// Kind:       backupLocation.Kind,
+					APIVersion: "stork.libopenstorage.org/v1alpha1",
+					Kind:       "BackupLocation",
+					Namespace:  backupLocation.Namespace,
+					Name:       backupLocation.Name,
+				}
+				resourceExport.Spec.TriggeredFrom = kdmputils.TriggeredFromStork
+				storkPodNs, err := k8sutils.GetStorkPodNamespace()
+				if err != nil {
+					logrus.Errorf("error in getting stork pod namespace: %v", err)
+					return err
+				}
+				resourceExport.Spec.TriggeredFromNs = storkPodNs
+				resourceExport.Spec.Source = *source
+				resourceExport.Spec.Destination = *destination
+
+				_, err = kdmpShedOps.Instance().CreateResourceExport(resourceExport)
+				if err != nil {
+					logrus.Errorf("failed to create DataExport CR: %v", err)
```

Comment:
Typo: DataExport - ResourceExport
Also it will be good to add RE name and ns in the error message